posts/2020-10-23-beta-20.0.0.12.adoc

```diff
@@ -0,0 +1,440 @@
+
+[#lra]
+=== Support for MicroProfile LRA
+
+Long Running Actions (LRA) provide a cloud native transaction model based on a compensating Saga system. This allows loosely coupled services to coordinate long running activities, guaranteeing eventual data consistency without the need to put locks on data.
+
+The features provide a Coordinator, to which LRAs along with any corresponding method are registered, and the participant which allows annotated microservice methods to create a new or join an existing LRA.
+
+To start or join an LRA use the `@LRA` annotation. Depending on the type provided it may create a new LRA, join the existing LRA or continue outside one if none is provided. A full list of options can be found in the Microprofile Long Running Actions link:https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc[specifications].
+
+An example of always creating a new LRA when called would be:
+
+[source, java]
+----
+    @LRA(LRA.Type.REQUIRES_NEW)
+    @Path(“/start”)
+    @PUT
+    public void startAction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
+    {
+        /*
+         * Business logic that may call upon other services to participate in this Long Running Action
+         */
+         return Response.ok().build();
+    } 
+----
+
+Once the LRA has been successfully finished all participating services will have their `@Complete` annotated methods called to perform the necessary cleanup:
+
+[source, java]
+----
+    @Complete
+    @Path("/complete")
+    @PUT
+    public Response completeAction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
+    {
+        /*
+         * Business logic to cleanup resources once the LRA has closed successfully
+         */
+         return Response.ok().build();
+    }  
+----
+
+Should a service in an LRA be canceled or timeout, the `@Compensate` methods for each participating service in the LRA will be called:
+
+[source, java]
+----
+    @Compensate
+    @Path("/compensate")
+    @PUT
+    public Response compensateAction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
+    {
+        /*
+         * As the LRA has been canceled business logic to compensate any changes is necessary
+         */
+         return Response.ok().build();
+    } 
+----
+
+To enable the LRA Coordinator feature in your app, add the following to your `server.xml`:
+[source, xml]
+----
+<featureManager>
+    <feature>cdi-2.0</feature>
+    <feature>jaxrs-2.1</feature>
+    <feature>mpLRACoordinator-1.0</feature>
+</featureManager>
+----
+
+To enable the LRA Participants feature in your app, add the following to your `server.xml`:
+[source, xml]
+----
+<featureManager>
+    <feature>cdi-2.0</feature>
+    <feature>jaxrs-2.1</feature>
+    <feature>mpLRA-1.0</feature>
+</featureManager>
+----
+
+For more information:
```

Comment:
It might also be nice to mention that MP LRA is not part of the `microprofile-4.0` feature because it is a standalone specification project.  So unlike the other MP technologies mentioned above, users will need to explicitly add `mpLRA-1.0` or `mpLRACoordinator-1.0` to use it.

Reply:
Added a small note to make the reader aware of this.